File: src/test/java/com/mbayou/kick4k/events/ChannelFollowedEventTest.java
package com.mbayou.kick4k.events;

import org.junit.jupiter.api.Test;
import com.mbayou.kick4k.events.type.ChannelFollowedEvent;
import com.mbayou.kick4k.events.type.EventUser;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ChannelFollowedEventTest extends KickEventDispatcherTest {
    private static final String PAYLOAD = """
            {
              "broadcaster": {
                "is_anonymous": false,
                "user_id": 123456789,
                "username": "broadcaster_name",
                "is_verified": true,
                "profile_picture": "https://example.com/broadcaster_avatar.jpg",
                "channel_slug": "broadcaster_channel",
                "identity": null
              },
              "follower": {
                "is_anonymous": false,
                "user_id": 987654321,
                "username": "follower_name",
                "is_verified": false,
                "profile_picture": "https://example.com/sender_avatar.jpg",
                "channel_slug": "follower_channel",
                "identity": null
              }
            }""";

    @Test
    public void should_deserialize_channel_followed_event() {
        ChannelFollowedEvent event = dispatchAndCapture(
                ChannelFollowedEvent.class,
                "channel.followed",
                "1",
                PAYLOAD
        );

        EventUser broadcaster = event.getBroadcaster();
        assertNotNull(broadcaster);
        assertFalse(broadcaster.getAnonymous());
        assertEquals(123456789, broadcaster.getUserId());
        assertEquals("broadcaster_name", broadcaster.getUsername());
        assertTrue(broadcaster.getVerified());
        assertEquals("https://example.com/broadcaster_avatar.jpg", broadcaster.getProfilePicture());
        assertEquals("broadcaster_channel", broadcaster.getChannelSlug());
        assertNull(broadcaster.getIdentity());

        EventUser follower = event.getFollower();
        assertNotNull(follower);
        assertFalse(follower.getAnonymous());
        assertEquals(987654321, follower.getUserId());
        assertEquals("follower_name", follower.getUsername());
        assertFalse(follower.getVerified());
        assertEquals("https://example.com/sender_avatar.jpg", follower.getProfilePicture());
        assertEquals("follower_channel", follower.getChannelSlug());
        assertNull(follower.getIdentity());
    }
}

File: src/test/kotlin/com/mbayou/kick4k/events/ChannelFollowedEventTest.kt
package com.mbayou.kick4k.events

import com.mbayou.kick4k.events.type.ChannelFollowedEvent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ChannelFollowedEventTest : KickEventDispatcherTest() {
    @Test
    fun shouldDeserializeChannelFollowedEvent() {
        val event = dispatchAndCapture(
            ChannelFollowedEvent::class.java,
            "channel.followed",
            "1",
            PAYLOAD,
        )

        val broadcaster = event.broadcaster
        assertNotNull(broadcaster)
        assertFalse(broadcaster!!.isAnonymous!!)
        assertEquals(123456789, broadcaster.userId)
        assertEquals("broadcaster_name", broadcaster.username)
        assertTrue(broadcaster.isVerified!!)
        assertEquals("https://example.com/broadcaster_avatar.jpg", broadcaster.profilePicture)
        assertEquals("broadcaster_channel", broadcaster.channelSlug)
        assertNull(broadcaster.identity)

        val follower = event.follower
        assertNotNull(follower)
        assertFalse(follower!!.isAnonymous!!)
        assertEquals(987654321, follower.userId)
        assertEquals("follower_name", follower.username)
        assertFalse(follower.isVerified!!)
        assertEquals("https://example.com/sender_avatar.jpg", follower.profilePicture)
        assertEquals("follower_channel", follower.channelSlug)
        assertNull(follower.identity)
    }

    companion object {
        private const val PAYLOAD = """
            {
              "broadcaster": {
                "is_anonymous": false,
                "user_id": 123456789,
                "username": "broadcaster_name",
                "is_verified": true,
                "profile_picture": "https://example.com/broadcaster_avatar.jpg",
                "channel_slug": "broadcaster_channel",
                "identity": null
              },
              "follower": {
                "is_anonymous": false,
                "user_id": 987654321,
                "username": "follower_name",
                "is_verified": false,
                "profile_picture": "https://example.com/sender_avatar.jpg",
                "channel_slug": "follower_channel",
                "identity": null
              }
            }
        """
    }
}
